src/gordon_gcp/clients/gdns.py

```diff
@@ -97,47 +97,111 @@ class GDNSClient(http.AIOConnection):
             object attached to :obj:`auth_client` if not provided.
     """
     BASE_URL = 'https://www.googleapis.com/dns'
+    # see https://cloud.google.com/dns/api/v1/changes#resource
+    DNS_CHANGES_DONE = 'done'
+    REVERSE_PREFIX = 'reverse-'
 
     def __init__(self, project=None, auth_client=None, api_version='v1',
                  session=None):
         super().__init__(auth_client=auth_client, session=session)
         self.project = project
         self._base_url = f'{self.BASE_URL}/{api_version}/projects/{project}'
 
-    def _parse_resp_to_records(self, response, records):
-        unparsed_records = response.get('rrsets', [])
-        for record in unparsed_records:
-            rrset = GCPResourceRecordSet(**record)
-            records.append(rrset)
+    @staticmethod
+    def get_rrsets_as_objects(rrsets):
+        """Return a list of rrsets as GCPResourceRecordSets.
+
+        Args:
+            rrsets (list of dict): RRsets represented as dicts.
+        Returns:
+            list of :class:`GCPResourceRecordSet` objects.
+        """
+        return [GCPResourceRecordSet(**rrset) for rrset in rrsets]
+
+    def get_managed_zone(self, zone):
+        """Get the GDNS managed zone name for a DNS zone.
+
+        Google uses custom string names with specific `requirements
+        <https://cloud.google.com/dns/api/v1/managedZones#resource>`_
+        for storing records. The scheme implemented here chooses a
+        managed zone name which removes the trailing dot and replaces
+        other dots with dashes, and in the case of reverse records,
+        uses only the two most significant octets, prepended with
+        'reverse'.
+
+        Example:
+           get_managed_zone('example.com.') = 'example-com'
+           get_managed_zone('30.20.10.in-addr.arpa.) = 'reverse-20-10'
+
+        Args:
+            zone (str): DNS zone.
+        Returns:
+            str of managed zone name.
+
+        """
+        if zone.endswith('.in-addr.arpa.'):
+            return self.REVERSE_PREFIX + '-'.join(zone.split('.')[-5:-3])
+        return '-'.join(zone.split('.')[:-1])
 
-    async def get_records_for_zone(self, zone):
-        """Get all resource record sets for a particular managed zone.
+    async def get_records_for_zone(self, dns_zone, params=None):
+        """Get all resource record sets for a managed zone, using the DNS zone.
 
         Args:
-            zone (str): Desired managed zone to query.
+            dns_zone (str): Desired DNS zone to query.
+            params (dict): Optional set of search parameters to pass to GDNS.
```

Comment:
I appreciate a little more clarity: these aren't search parameters (it's just the calling method refers to them as such), they are query parameters. Perhaps something like 

```
params (dict): (optional) Additional query parameters for HTTP requests to the GDNS API.
```

(keeping in line with how we typically [denote optional params](https://github.com/spotify/gordon-gcp/search?q=optional&unscoped_q=optional))

Reply:
ok